src/HttpHelper/HttpClientHelper/HttpClient.php

```diff
@@ -7,80 +7,111 @@
+    /**
+     * @throws GuzzleHttp\Exception\GuzzleException
+     */
+    public function patch(string $sUrl, array $aRequestOptions, string $sAction, LogContext $logContext, array $aLogMetadata = []): ResponseInterface
     {
-        $aLogMetadata['url'] = $sUrl;
-        $aLogMetadata['httpCode'] = $response->getHttpCode();
-        $aLogMetadata['response'] = $response->getResponse();
-        $request && $aLogMetadata['request'] = $request->getBody();
-
-        if (!$response->isSuccess()) {
-            Logger::error("$sAction has failed: {$response->getErrorMessage()}", $logContext, $aLogMetadata);
-        } else {
-            Logger::debug("$sAction succeed", $logContext, $aLogMetadata);
-        }
+        return $this->request(Request::METHOD_PATCH, $sUrl, $aRequestOptions, $sAction, $logContext, $aLogMetadata);
     }
 
-    private function request(string $sMethod, string $sUrl, array $aRequestOptions, $fRequestTimeout = null): HttpClientResponse
+    private function request(
+        string $sMethod,
+        string $sUrl,
+        array $aRequestOptions,
+        string $sAction,
+        LogContext $logContext,
+        array $aLogMetadata = []
+    ): ResponseInterface
     {
-        $bError = true;
         try {
-            $aRequestOptions[RequestOptions::TIMEOUT] = $fRequestTimeout ? : self::DEFAULT_REQUEST_TIMEOUT;
+            $aRequestOptions[RequestOptions::TIMEOUT] = $aRequestOptions[RequestOptions::TIMEOUT] ?? self::DEFAULT_REQUEST_TIMEOUT;
             $response = $this->httpClient->request($sMethod, $sUrl, $aRequestOptions);
+
             $sResponseBody = $response->getBody()->getContents();
-            $iHttpCode = $response->getStatusCode();
-            $bError = false;
-        } catch (ConnectException $e) {
-            $sResponseBody = "ConnectException: {$e->getMessage()}";
-            $iHttpCode = 0;
+            $response->getBody()->rewind();
+            Logger::debug("$sAction : succeed requesting $sMethod $sUrl - {$response->getStatusCode()}", $logContext, \array_merge($aLogMetadata, [
+                'response' => $sResponseBody,
+                'requestOptions' => $this->hideRequestSecrets($aRequestOptions)
+            ]));
         } catch (GuzzleException $e) {
-            if (\method_exists($e, 'hasResponse') && $e->hasResponse()) {
-                $response = $e->getResponse();
-                $sResponseBody = $response->getBody()->getContents();
-                $iHttpCode = $response->getStatusCode();
-            } else {
-                $sResponseBody = "GuzzleException: {$e->getMessage()}";
-                $iHttpCode = 0;
-            }
+            Logger::debug("$sAction : error while requesting $sMethod $sUrl - {$e->getCode()}", $logContext, \array_merge($aLogMetadata, [
+                'response' => $e->getMessage(),
+                'requestOptions' => $this->hideRequestSecrets($aRequestOptions)
+            ]));
+
+            throw $e;
```

Comment:
Same than https://github.com/proleadin/symfony-survival-kit/pull/6/files maybe not the same version of guzzle so maybe treatment is going to be different